tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/BinaryPrimitivesHelperTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="BinaryPrimitivesHelperTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DataStreamsMonitoring.Utils;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class BinaryPrimitivesHelperTests
+{
+    [Fact]
+    public void CanRoundTripValue()
```

Comment:
Can you also write the following tests:
- Generate a random `ulong` number (or I suppose a hard-coded number would work as well), zero out the lowest byte, write to the byte buffer using `WriteUInt64LittleEndian(bytes, expected)`, assert that `bytes[0] == 0`
- Generate a random 8-bit byte array (or I suppose a hard-coded array would work as well), zero out `bytes[0]`, read from the byte buffer using `ReadUInt64LittleEndian(bytes)`, assert that the lowest byte is `0`

Reply:
I added this in https://github.com/DataDog/dd-trace-dotnet/pull/3131/commits/3e94261ebdca0e7b19a7febff6c355d9aaaa328c, but I don't think it actually satisfies your concerns, as we never test the big-endian code! 🙂

Reply:
Do you think you could introduce a branch in the new test, one for little-endian and one for big-endian and inside each branch assert that the "lowest" byte is 0? I'm not sure when it would get hit, but if we were to run it on a big-endian system and it failed that would be a red flag

Reply:
> Do you think you could introduce a branch in the new test, one for little-endian and one for big-endian and inside each branch assert that the "lowest" byte is 0?

Not _really_ 😄  There's no way to hit the "big-endian" branch without running on a big-endian machine. And if we _did_, then this test _would_ be testing the big-endian code path (instead of the little endian) so that would be your red flag!